### PR TITLE
Finish cloud login flow

### PR DESCRIPTION
**Pull request by balloob**
- Add flow for registration and forgot password
- Add new layout partial `hass-subpage` that renders a header with a back button. Used all over the config panel.
- Navigate mixin can now also replace the current url instead of pushing new one on the stack
- Add CSS to make a button look like a link

### CODE REVIEW

File: panels/config/cloud/ha-config-cloud-account.html [hunk 1]
```diff
@@ -57,6 +77,10 @@
     this.hass.callApi('post', 'cloud/logout').then(
       () => this.fire('ha-account-refreshed', { account: null }));
   }
+
+  _backTapped() {
```
Comment (andrey-git) [inline panels/config/cloud/ha-config-cloud-account.html]: This is not needed

**Review by andrey-git — COMMENTED** [review]
I think the use of iron-pages is problematic.

While it could be desired to load all tabs of `states` so switching tabs is fast, it makes loading the config page very slow, as it renders each individual config subpage and each of those makes an API calls.

I think `dom-if` while less compact is a better approach for both root of config and maybe the cloud config subpage.

**Comment by balloob** [comment]
I'll upgrade the config panel to use `dom-if` exclusively.

**Comment by balloob** [comment]
I've addressed your comments. Left the `iron-pages` for the config panel root for now.

**Review by andrey-git — APPROVED** [review]

**Pull request merged**